manual/CHAPTER_TextRtlil.tex

```diff
@@ -0,0 +1,299 @@
+\chapter{RTLIL Text Representation}
+\label{chapter:textrtlil}
+
+% Stolen from stackexchange: calculate indent based on given keyword,
+% with some nice hrules added in.
+\newlength{\myl}
+
+\newenvironment{indentgrammar}[1]
+    {\vspace{0.5cm}\hrule
+    \setlength{\myl}{\widthof{#1}+2em}
+    \grammarindent\the\myl
+    \begin{grammar}}
+    {\end{grammar}
+    \hrule}
+
+This appendix documents the text representation of RTLIL in extended Backus-Naur form (EBNF).
+
+The grammar is not meant to represent semantic limitations. That is, the grammar is ``permissive'', and later stages of processing perform more rigorous checks.
+
+The grammar is also not meant to represent the exact grammar used in the RTLIL frontend, since that grammar is specific to processing by lex and yacc, is even more permissive, and is somewhat less understandable than simple EBNF notation.
+
+Finally, note that all statements (rules ending in \texttt{-stmt}) terminate in an end-of-line. Because of this, a statement cannot be broken into multiple lines.
+
+\section{Lexical elements}
+
+\subsection{Characters}
+
+The characters accepted in an RTLIL file are those encodable in 8 bits. Unicode is not supported. For maximum safety, limit characters to the 7-bit ASCII range $[0,127]$.
```

Comment:
Wait, is there anything in Yosys that actually cares about 8-bit characters? The low block (0x20 and below) are forbidden in identifiers, but I don't think the high half is.

Reply:
Anecdotally, I haven't had a problem with UTF-8 (provided you use an escaped identifier in the case of Verilog, anyway)

Reply:
Hmm, I should probably change this. I originally meant to say that you can't use Unicode, and if you do use a multi-char codepoint, it will just look like a bunch of 8-bit characters. Let me fix this.

Reply:
Ah, I found the assert in rtlil.h that prohibits anything 0x20 and below from being in identifiers.

Reply:
Done. I've clarified the section on characters.